File: ice0208/hash-map_set/2215.py
class Solution:
    def findDifference(self, nums1: list[int], nums2: list[int]) -> list[list[int]]:
        # set을 이용하여 중복을 제거후, 차집합으로 정답 구하기

        answer = [list(set(nums1) - set(nums2)), list(set(nums2) - set(nums1))]
        return answer


if __name__ == "__main__":
    print(Solution().findDifference(nums1=[1, 2, 3], nums2=[2, 4, 6]))
    print(Solution().findDifference(nums1=[1, 2, 3, 3], nums2=[1, 1, 2, 2]))

File: moonpoison/hash_map_set/2215.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) 
    {
        unordered_set<int> s1(nums1.begin(), nums1.end()), s2(nums2.begin(), nums2.end());
        vector<int> ans1, ans2;
        for(int i:s1)
            if(s2.count(i)==0) ans1.push_back(i);
        for(int i:s2)
            if(s1.count(i)==0) ans2.push_back(i);
        return {ans1, ans2};
    }
};